src/main/java/in/projecteka/consentmanager/consent/ConsentManager.java

```diff
@@ -0,0 +1,73 @@
+package in.projecteka.consentmanager.consent;
+
+import in.projecteka.consentmanager.clients.ClientRegistryClient;
+import in.projecteka.consentmanager.clients.UserServiceClient;
+import in.projecteka.consentmanager.consent.model.request.ConsentDetail;
+import in.projecteka.consentmanager.consent.repository.ConsentRequestRepository;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+import java.util.UUID;
+
+public class ConsentManager {
+
+    private final ClientRegistryClient providerClient;
+    private UserServiceClient userServiceClient;
+    private final ConsentRequestRepository repository;
+
+    public ConsentManager(ConsentRequestRepository requestRepository, ClientRegistryClient providerClient, UserServiceClient userServiceClient) {
```

Comment:
regarding validations - IMHO, there are validations of different types and ought to be done in different places. I am currently using JSR303 spring validation which happens during the binding time. This just validates the schema constraints. 

Then there are entity and/or biz validations - which must happen from the service context level. Example - is the HIU valid, is there enough time for the transfer (if the dateExpiry is too close) etc etc. 

Sure, we can make the parallel calls using Mono better. I would say that we target later. Currently if any of the calls fail, its ok .. because all the calls to provider must succeed in this case.